### ISSUE TITLE
HTTP Client using curl.

### ISSUE THREAD

**Issue by h4cc** [doctrine/couchdb-client#38]
I was wondering why there is no client using curl, guzzle or buzz?

By using Guzzle even a async response handling could be done: http://guzzle.readthedocs.org/en/latest/clients.html#asynchronous-requests

Should such clients be part of this package, or should it be a external package?

**Comment by beberlei** [comment]
I would say a cURL based solution would be ok in core, Guzzle or Buzz no. The simplicity of the whole thing doesn't make it difficult to use curl. We cannot use the async response handling anyways, because most of the code is directly querying the response object and we cant make it lazy.

**Comment by robsonvn** [comment]
I agree with @beberlei, in addition, I would say the using cURL would be a lot better for maintenance and to keep the code cleaner and tidier. 

I'll analyse the impacts of moving to cURL soon to see if the advantages will outweigh the disadvantages (rework)
